app/models/calendar.rb

```diff
@@ -0,0 +1,32 @@
+require "icalendar"
+
+class Calendar
+  def initialize(events)
+    @events = events
+  end
+
+  def publish
+    calendar = Icalendar::Calendar.new
+
+    calendar.timezone do |timezone|
+      timezone.tzid = "UTC"
+    end
+
+    @events.each do |event|
+      calendar.event do |e|
+        e.dtstart     = Icalendar::Values::DateTime.new(event.start_at.utc)
+        e.dtend       = Icalendar::Values::DateTime.new((event.start_at + 2.hours).utc)
+        e.summary     = event.name
+        e.description = event.presentation
+        e.location    = "Toronto, Canada" # TODO: Use event.location, but it's too long for the calendar with all the details and instructions
```

Comment:
I hard-coded `"Toronto, Canada"` for now, but ideally we could use the `event.location` value. The problem is that the `event.location` has too much information. Maybe we could add a `city` field to the `Event` model that would use for this and then have the `event.location` information as part of the Ical event `description`.

Reply:
Does the conferences site use this information? My hesitation with adding a `city` field is that it will probably always be `Toronto, Canada`, so I'm comfortable with this for now.

Reply:
Yeah, it's what we show for this field here:

![CleanShot 2024-09-03 at 01 20 02](https://github.com/user-attachments/assets/c365a8cb-b444-4f3a-8ee8-b2bb7d1efc4d)

It wouldn't make sense for us to show anything other than the city on the meetups page, but it totally makes sense for your use case.

Reply:
Okay, I just added a field to the db which defaults to "Toronto, Canada", just so we can remove the TODO and give you the option to host an event outside of Toronto ;)